scripts/common.py

```diff
@@ -437,6 +463,23 @@ def convert_to_posixpath(file_path):
     return file_path.replace('\\', '/')
 
 
+def inplace_replace_file(filename, regex_pattern, replace):
+    """Replace the file content inplace with regex pattern.
```

Comment:
Just to clarify: is the idea that, you are looking at the file line by line, and for each line, you are finding all matches for the regex, and replacing them with the new string?

If so, change the arg `replace` to `replacement_string`, since replace sounds like a function name. Also the docstring should be clearer: "Finds all instances of regex_pattern in the given file and replaces them with replacement_string."

Please also document that the regex only works for stuff that is in the same line of code (i.e. regexes containing newlines aren't going to work -- though I wonder why that is? It would be better not to do things that way, IMO, and process the entire file as a string). Perhaps this is something that could be made part of the function name as well.

Also document what happens if regex finds are overlapping. E.g. file contains "aaaaaaa" and regex is "aa" and replacement is "b". There are 6 instances of "aa" in the file; which of those get replaced?

Reply:
Also if you can simplify this function enough, then you can probably just inline it.

Reply:
I did not expect this function to achieve such a comprehensive job. This functionality can be found at many places as inline code across the scripts runner code base, so I thought it maybe better to provide a function (also when writing unit test, I just need to mock this function instead of a lot of ones).

As far as I can see, this function in the code base is just doing some trivia jobs, so I did not consider so many cases. If you feel it is better to inline it, I will just inline it.
https://github.com/oppia/oppia/blob/1f4b118c7b69d23439599389537d7db19180979b/scripts/start.py#L106
 https://github.com/oppia/oppia/blob/1f4b118c7b69d23439599389537d7db19180979b/scripts/start.py#L117
https://github.com/oppia/oppia/blob/1f4b118c7b69d23439599389537d7db19180979b/scripts/create_expression_parser.py#L54
https://github.com/oppia/oppia/blob/1f4b118c7b69d23439599389537d7db19180979b/scripts/create_expression_parser.py#L65

Reply:
Ah, thanks for noticing the duplication. I think it's fine to create a utility function if its contract is defined really clearly. In particular, it is OK to stipulate that the input regex is applied to the file line by line (so I retract my parenthetical comment in the third paragraph).

Reply:
Done